internal/oci/container.go

```diff
@@ -50,25 +50,29 @@ type Container struct {
 	dir        string
 	stopSignal string
 	// If set, _some_ name of the image imageID; it may have NO RELATIONSHIP to the users’ requested image name.
-	someNameOfTheImage    *references.RegistryImageReference
-	imageID               *storage.StorageImageID // nil for infra containers.
-	mountPoint            string
-	seccompProfilePath    string
-	conmonCgroupfsPath    string
-	crioAnnotations       fields.Set
-	state                 *ContainerState
-	opLock                sync.RWMutex
-	spec                  *specs.Spec
-	idMappings            *idtools.IDMappings
-	terminal              bool
-	stdin                 bool
-	stdinOnce             bool
-	created               bool
-	spoofed               bool
-	stopping              bool
-	stopLock              sync.Mutex
+	someNameOfTheImage *references.RegistryImageReference
+	imageID            *storage.StorageImageID // nil for infra containers.
+	mountPoint         string
+	seccompProfilePath string
+	conmonCgroupfsPath string
+	crioAnnotations    fields.Set
+	state              *ContainerState
+	opLock             sync.RWMutex
+	spec               *specs.Spec
+	idMappings         *idtools.IDMappings
+	terminal           bool
+	stdin              bool
+	stdinOnce          bool
+	created            bool
+	spoofed            bool
+	stopping           bool
+	stopLock           sync.Mutex
+	// stopTimeoutChan is used to update the stop timeout.
+	// After the container goes into the kill loop, the channel must not be used
+	// because it is not controlled by the timeout anymore.
 	stopTimeoutChan       chan int64
 	stopWatchers          []chan struct{}
+	stopKillLoop          bool
```

Comment:
nit: maybe stopKillLoopBegun or something? making clear it's part of the 'stop' operation and not an instruction whether to stop the kill loop

Reply:
Thanks! done